samples/bluetooth/peripheral_hids/src/main.c

```diff
@@ -58,13 +58,19 @@ static void disconnected(struct bt_conn *conn, u8_t reason)
 	printk("Disconnected from %s (reason 0x%02x)\n", addr, reason);
 }
 
-static void security_changed(struct bt_conn *conn, bt_security_t level)
+static void security_changed(struct bt_conn *conn, bt_security_t level,
+			     u8_t status)
 {
 	char addr[BT_ADDR_LE_STR_LEN];
 
 	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
-	printk("Security changed: %s level %u\n", addr, level);
+	if (!status) {
+		printk("Security changed: %s level %u", addr, level);
+	} else {
+		printk("Security failed: %s level %u reason %d", addr, level,
+		       status);
```

Comment:
Why "reason" in the log string? We now have "reason", "status" and "err" for the same thing. Let's try to stick to just one if possible.

Reply:
I suggest `status` because the enum contains a `SUCCESS`

Reply:
Then the names of the enum values should also need to be changed to `BT_SECURITY_STATUS_...` which makes them rather long (well, just three characters, but the fact that we originally didn't start truncating "security" to "sec" doesn't help). I'm more or less fine either way, perhaps slightly leaning toward `err` since we do already use it elsewhere where "success" is possible. E.g. in the bt_ready and connected callbacks.

Reply:
We also have BT_HCI_ERR_SUCCESS. In the connected callback we give this parameter as "err", in disconnected it is "reason" (which makes sense since it is never 0 here)
I suggest to keep this in sync with HCI error codes and use 'err' for the security_changed, and as `reason` in pairing_failed.